### PR TITLE
chore: add job ID and experiment labels to prometheus endpoint mappings [DET-6964]

### PR DESCRIPTION
**Pull request by azhou-determined**
## Description
Updates to prometheus endpoint:
- add job ID to det-state mappings
- add experiment labels on experiment creation

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/prom/prom.go [hunk 1]
```diff
@@ -25,14 +26,20 @@ Exposes mapping of allocation ID to container ID`,
...
 	containerIDToRuntimeID = promauto.NewGaugeVec(prometheus.GaugeOpts{
 		Subsystem: "det",
 		Name:      "container_id_runtime_container_id",
 		Help:      "a mapping of the container ID to the container ID given be the runtime",
 	}, []string{"container_runtime_id", "container_id"})
 
+	taskIDToExperimentID = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Subsystem: "det",
+		Name:      "task_id_experiment_id",
+		Help:      "a mapping of the experiment ID to the labels",
```
Comment (stoksc) [inline master/internal/prom/prom.go:40]:
```suggestion
		Help:      "a mapping of the task ID to the experiment ID",
```

**Review by stoksc — COMMENTED** [review]
what's our strategy on many map[x]y series vs (almost just) one map[x]map[y]map[z]... series? I'm not sure what is better. (though queries are way easier with the latter)

**Comment by stoksc** [comment]
also, why `task_id_experiment_id` as opposed to `job_id_experiment_id`

**Comment by azhou-determined** [comment]
> why task_id_experiment_id as opposed to job_id_experiment_id

no good reason. made the changes before adding job_id to the mappings. is job_id a better mapping? and why? 

> what's our strategy on many map[x]y series vs (almost just) one map[x]map[y]map[z]... series? I'm not sure what is better. (though queries are way easier with the latter)

yeah, it's a tradeoff between easier querying and easier code readability/cleanliness. doing a single mapping would require pulling in random data from various parts of the backend. also, a lot of queries are specific to filters and don't need to join all of the mappings (like experiment labels, jobs, etc) so it seems more straightforward as a strategy to try and add what we need piece-by-piece.

**Comment by stoksc** [comment]
> no good reason. made the changes before adding job_id to the mappings. is job_id a better mapping? and why?

mostly just how the hierarchy of things works, it makes more sense. but it's not the end of the world.

**Review by stoksc — APPROVED** [review]
lgtm, though I think job_id_experiment_id makes a lot more sense

**Comment by azhou-determined** [comment]
> lgtm, though I think job_id_experiment_id makes a lot more sense

made the change. 👍

**Pull request merged**